packages/eas-cli/src/authUtils.ts

```diff
@@ -0,0 +1,21 @@
+import SessionManager from './user/SessionManager';
+
+// Handle the case where a user needs to be in sudo mode to perform an action.
+export async function withSudoModeAsync<T>(
+  sessionManager: SessionManager,
+  graphqlQuery: () => Promise<T>
+): Promise<T> {
+  try {
+    return await graphqlQuery();
+  } catch (error: any) {
+    const isSudoError = error.graphQLErrors?.some(
+      (err: { extensions: { errorCode: string } }) =>
+        err?.extensions?.errorCode === 'SUDO_MODE_REQUIRED'
+    );
+    if (!isSudoError) {
+      throw error;
+    }
+    await sessionManager.showSudoPromptAsync({ sso: false });
+    return await graphqlQuery();
+  }
+}
```

Comment:
Offline discussion feedback:

It seems like requiring sudo mode is problematic for now, it doesn't work for SSO auth and doesn't seem to work with `--non-interactive` mode, which is a big use case. For now, we decided to allow people to read sensitive env vars without sudo mode. This will require backend changes + dropping sudo logic from this PR.